### PR TITLE
Gradle: Fail nicely if an unsupported Gradle version < 3.3 is used

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Check the Gradle version of the project first and return an error message
if it is unsupported instead of throwing an exception later on.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/414)

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/gradle-unsupported-version/settings.gradle [hunk 1]
```diff
@@ -0,0 +1 @@
+// Required to prevent Gradle from reading the settings.gradle from the root directory of the ORT project.
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gradle-unsupported-version/settings.gradle]: Nit: No final line ending.

File: analyzer/src/main/resources/init.gradle [hunk 2]
```diff
@@ -113,6 +113,17 @@ class DependencyTreePlugin implements Plugin<Gradle> {
 
             List<String> errors = []
 
+            def gradleVersion = project.gradle.gradleVersion.split('\\.')
+
+            if (gradleVersion[0].toInteger() < 3
+                    || (gradleVersion[0].toInteger() == 3 && gradleVersion[1].toInteger() < 3)) {
+                return new DependencyTreeModelImpl(project.group ?: '', project.name, project.version ?: '', [], [],
+                        [
```
Comment (sschuberth) [inline analyzer/src/main/resources/init.gradle]: Nit: Isn't the formatting a bit weird here? I would have expected `[` / `]` to direct go before / after the inner `(` / `)`. Just wondering, not important anyway.

File: analyzer/src/funTest/assets/projects/synthetic/gradle-unsupported-version/settings.gradle (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/gradle-unsupported-version/settings.gradle]: Done.

File: analyzer/src/main/resources/init.gradle (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/main/resources/init.gradle]: Done.

**Pull request merged**